### PR TITLE
Update ScriptCache to clear ATP assets when disconnected from a domain

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: decb94b719311dba73338538f73216214c6fc37b available at: https://gist.github.com/85ac2f6bab519cbe1b121b13b2987da0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8806-decb94b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8806-decb94b.dmg

**Comment by huffman** [comment]
With this PR, the ScriptCache will clear ATP assets from it's cache when getting disconnected from a domain.

To test:
 1. Go to your domain
 1. Upload the following script to your asset server as "/prtest_atptesting.js": https://gist.githubusercontent.com/huffman/c9903ab22778b753e5f87f2246859452/raw/6d211c94918f5bed981bad03c81fed7379d394d6/prtest_atptesting_1.js
 1. Run https://gist.githubusercontent.com/huffman/0b511df298738afefd5ccf61507a4e63/raw/113b9cd30c039867596e05fe1026ef3acf4fd267/prtest_atptesting_client.js
 1. You should see something like this in your log: `[10/13 14:25:26] [DEBUG] [hifi.scriptengine] script:print()<< prtest_atptesting_1.js`
 1. Upload the following script to your asset server as "/prtest_atptesting.js", overwriting the previous asset: https://gist.githubusercontent.com/huffman/96bcb0f4ef534fc270367bfd481f8062/raw/68d8dd8398a85d14a242d06c9530e5cf3ced67f2/prtest_atptesting_2.js
 1. Rerun `prtest_atptesting_client.js`, you should see the same output as before
 1. Shut your domain down, once it is down, bring it back up
 1. Rerun `prtest_atptesting_client.js`, you should now see something like: `[10/13 14:25:26] [DEBUG] [hifi.scriptengine] script:print()<< prtest_atptesting_2.js`

__Part 2__ Redo the above steps, but in step 7, instead of shutting your down down, go to a different domain and then go back to your domain.

**Comment by cozza13** [comment]
Confirmed that on shut down and domain movement that the cache refreshed.

File: libraries/script-engine/src/ScriptCache.cpp [hunk 1]
```diff
@@ -36,6 +36,19 @@ void ScriptCache::clearCache() {
     _scriptCache.clear();
 }
 
+void ScriptCache::clearATPScriptsFromCache() {
+    Lock lock(_containerLock);
+    qDebug(scriptengine) << "Clearing ATP scripts from ScriptCache";
+    for (auto it = _scriptCache.begin(); it != _scriptCache.end();) {
+        if (it.key().scheme() == "atp") {
+            qDebug(scriptengine) << "Removing: " << it.key();
```
Comment (birarda) [inline libraries/script-engine/src/ScriptCache.cpp]: what's the difference between `qDebug` and `qCDebug`?

Comment (huffman) [inline libraries/script-engine/src/ScriptCache.cpp]: That was a typo, I'm surprised you can do that. I will switch it.

Comment (huffman) [inline libraries/script-engine/src/ScriptCache.cpp]: Updated

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7626f49ddc964b4a1d69609f87fd4280f33b35ee available at: https://gist.github.com/85ac2f6bab519cbe1b121b13b2987da0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8806-7626f49.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8806-7626f49.dmg

**Pull request merged**